### PR TITLE
Add a command, to add commands

### PR DESCRIPTION
**Pull request by Waterdrips**
## Description

Closes https://github.com/alexellis/arkade/issues/379

This command pre-generates a default command with pre-filled values. A
simple chart app should be able to be added very quickly using the
template.

arkade new-app --name OpenFaaS --command openfaas

will generate openfaas_app.go, pre-fill names. You then need to add the
chart url, any extra options and register the command in
'cmd/install.go'
Signed-off-by: Alistair Hey <alistair@heyal.co.uk>

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

#379 

## How Has This Been Tested?

locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

File: cmd/new_app.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+type App struct {
+	Name         string
+	Command      string
+	ChartURL     string
+	HelmRepoName string
+	Year         int
+}
+
+var NewAppTemplate = `// Copyright (c) arkade author(s) {{.Year}}. All rights reserved.
```
Comment (alexellis) [inline cmd/new_app.go:78]: Could you make this a file on disk, which is bundled via Go 1.16's embedding capability?

File: cmd/new_app.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		if err != nil {
+			return err
+		}
+
+		f, err := os.Create(fmt.Sprintf("./cmd/apps/%s_app.go", command))
+		if err != nil {
+			return err
+		}
+
+		err = tmpl.Execute(f, app)
```
Comment (alexellis) [inline cmd/new_app.go:58]: Where does the output go? To a file, to stdout?

What about the next steps like adding it to the list of apps and the README? Can you print a help message for that?

**Comment by alexellis** [comment]
@dirien @developer-guy what do you think?

**Comment by dirien** [comment]
I like the idea, makes contributing easier.

Just one thought would be, to bring OpenFaaS closer to arkade (eat own dogfood :)): could we get the template delivered via a FaaS? So we could seperate "config" from code...

**Pull request closed without merging**